addons/portal/static/src/js/portal.js

```diff
@@ -49,7 +59,67 @@ publicWidget.registry.portalDetails = publicWidget.Widget.extend({
      * @private
      */
     _onCountryChange: function () {
-        this._adaptAddressForm();
+        this._adaptAddressForm(this._getCountryId());
+    },
+
+    _onClickConfirmMainAddress: async function (ev) {
+        ev.preventDefault();
+
+        let form = $(ev.currentTarget.form);
+        let data = form.serializeArray().reduce(function(data, {name, value}) {
+            if (name in data) {
+                if (Array.isArray(data[name])) {
+                    data[name].append(value);
+                } else {
+                    data[name] = [data[name], value];
+                }
+            } else {
+                data[name] = value;
+            }
+            return data;
+        }, {});
+        await this._rpc({
+            route: "/my/main_address_confirmation_warnings",
+            params: {
+                data: data,
+            },
+        }).then(warnings => {
+            if (!warnings.length) {
+                form.submit();
+            } else {
+                warnings = warnings.map(warning => warning.split("\n"));
+                let dialog = new Dialog(null, {
```

Comment:
I would pass the context of the current widget to the `Dialog` constructor so that the Dialog can communicate with its parent by using the build-in propagation mechanism (see: `trigger_up`).

```javascript
let dialog = new Dialog(this, {
    // ...
});
```